.github/workflows/smoke.yaml

```diff
@@ -0,0 +1,68 @@
+name: ops Smoke Tests
+
+on:
+  workflow_dispatch:
+    schedule:
+      - cron:  '0 7 25 * *'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # pylibjuju does not currently support Juju 4.x
+        # The smoke tests do not yet work on Juju 2.9.
+        juju-version: ['3.5']
+        charmcraft-version: ['2.x', '3.x']
+        cloud: ['lxd', 'microk8s']
+
+    env:
+      JUJU_VERSION: "${{ matrix.juju-version }}"
+
+    steps:
+      # LXD is required for charmcraft to pack, even if it's not used as the
+      # Juju cloud.
+      - name: Set up LXD
+        uses: canonical/setup-lxd@8fb85546a934dfb994becf81341dd387ffe6aabb
+        with:
+          channel: 5.0/stable
+
+      - name: Set up Microk8s
+        if: matrix.cloud == 'microk8s'
+        uses: balchua/microk8s-actions@v0.3.2
+        with:
+          channel: '1.26-strict/stable'
+          devMode: 'true'
+          addons: '["dns", "hostpath-storage"]'
+
+      - name: Set up Juju (classic)
+        if: matrix.juju-version == '2.9'
```

Comment:
Shall we just drop the "if 2.9" stuff here, seeing we're not running these on 2.9? Or do you think we should add that in the near future?

Reply:
Yeah, good question. I started out planning to do 2.9, 3.[latest stable], and 4.0, thinking that would be a good range of coverage. 2.9 because it's still in "security fix support" and seems to have non-trivial use still, 3.[latest stable] because it seems unlikely that we'd break deploying only for one 3.x, and 4.0 because it's good to be proactive. I then ran into a bunch of issues and ended up dropping down to just 3.x.

`tox -e smoke` passes for me (except for trying to deploy on 24.04) locally, so I think it shouldn't be too much work to get it working in CI as well. I'd like to use a quite small amount of time to try to get that going and if not dropping it would make sense yes. I could cut it from this PR and add it back later if that was preferred.

Reply:
Sounds good. I'm happy either way -- no specific urgency on this.